File: python/remove_duplicates.py
class Solution(object):
    # O(n) time | O(1) space
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i = 0
        j = 0

        while j < len(nums):
            nums[i] = nums[j]
            while j < len(nums) and nums[i] == nums[j]:
                j += 1
            i += 1
        return i

File: src/main/java/july_2025/RemoveDuplicates.java
package july_2025;

public class RemoveDuplicates {

    // O(n) time | O(1) space
    public int removeDuplicates(int[] nums) {
        int L = 0;
        int R = 0;

        while (R < nums.length) {
            nums[L] = nums[R];
            while (R < nums.length && nums[L] == nums[R]) {
                R++;
            }
            L++;
        }
        return L;
    }

}
